### PR TITLE
Also send reports for the enforced CSP

### PR DESCRIPTION
**Pull request by robhudson**
## One-line summary

Updates our CSP settings so that we send reports to Sentry for both the enforced and report-only policies. We want to know if the enforced policy ever gets triggered so we can fix as soon as possible.

## Testing

Add something like this to your local `.env` file:
```
CSP_REPORT_URI=http://localhost:8000/_csp
```
and verify the headers for `Content-Security-Policy` contain a `report-uri`.

### CODE REVIEW

**Review by pmac — APPROVED** [review]

File: bedrock/settings/__init__.py [hunk 1]
```diff
@@ -246,8 +246,10 @@
 csp_extra_frame_src = config("CSP_EXTRA_FRAME_SRC", default="", parser=ListOf(str))
 if csp_extra_frame_src:
     _csp_child_src = list(set(_csp_child_src + csp_extra_frame_src))
+csp_report_uri = config("CSP_REPORT_URI", default="") or None
 
 CONTENT_SECURITY_POLICY = {
+    "REPORT_PERCENTAGE": config("CSP_REPORT_PERCENTAGE", default="100", parser=int),
```
Comment (stevejalim) [inline bedrock/settings/__init__.py:252]: Looks like Dev is set to report 5% and stage 2%, but prod has no CSP_REPORT_PERCENTAGE set, which would default to 100% on rollout and could spam Sentry quite hard, maybe?

Might be worth making `default="1"` here, as well as setting a real value for prod in `webservices-infra`

**Review by stevejalim — CHANGES_REQUESTED** [review]

**Comment by robhudson** [comment]
Updated to default to 1 but also added `CSP_REPORT_PERCENTAGE` to prod in https://github.com/mozilla-it/webservices-infra/pull/2783

**Review by stevejalim — APPROVED** [review]
r+ cheers!

**Review by janbrasna — COMMENTED** [review]
Is this so that report-only and enforced rules set the same percentage being submitted, and use the same endpoint too? (Do they make sense inside Sentry, i.e. reasonably distinguishable if that came from CSP or CSP-RO?)

Also, float vs. int discrepancy:

File: bedrock/settings/__init__.py [hunk 2]
```diff
@@ -246,8 +246,11 @@
 csp_extra_frame_src = config("CSP_EXTRA_FRAME_SRC", default="", parser=ListOf(str))
 if csp_extra_frame_src:
     _csp_child_src = list(set(_csp_child_src + csp_extra_frame_src))
+csp_report_uri = config("CSP_REPORT_URI", default="") or None
 
 CONTENT_SECURITY_POLICY = {
+    # Default report percentage to 1% just in case the env var isn't set, we don't want to bombard Sentry.
+    "REPORT_PERCENTAGE": config("CSP_REPORT_PERCENTAGE", default="1", parser=int),
```
[the remark this answers is not in the document]
Comment (robhudson) [reply bedrock/settings/__init__.py:253]: Sorry I didn't see your questions. We did find that Sentry doesn't really easily distinguish between enforced vs report-only. They have a "disposition" field but you have to click into the report to see it, it's not distinguishable from the list of issues page unfortunately. These were also adding a lot of noise to our main sentry project so in a later commit I set things up so the report-only violations go to a separate sentry project. (I also noticed the int parser and fixed that there)

[the remark this answers is not in the document]
Comment (janbrasna) [reply bedrock/settings/__init__.py:253]: Good we're thinking the same;) Yes, they do collect the disposition, but truth is I wasn't even aware where to look for it (they avoid report-only pretty much in the docs…) — so it's not surprising it's there somewhere in the metadata, but hidden away in a way it's not useful to tell the different sources of reports easily apart. Good that you managed to set up a separate project to receive just the report-only violations (from both prod/stage & dev/demos; or you run report-only just for some of the environments?)

**Pull request merged**